docs/source/howto/workflows.rst

```diff
@@ -135,171 +293,170 @@ The second argument is the result of the work chain, extracted from the ``Int``
 
-Running a work function is as simple as calling a typical Python function: simply call it with the required input arguments:
+.. dropdown:: ``BadMultiplyAddIsEvenWorkChain``
 
-.. code-block:: python
+    .. literalinclude:: include/snippets/extend_workflows.py
+        :language: python
+        :pyobject: BadMultiplyAddIsEvenWorkChain
+.. note::
 
-    from aiida.plugins import WorkflowFactory, DataFactory
-    add_and_multiply = WorkflowFactory('arithmetic.add_multiply')
-    Int = DataFactory('int')
+    We've removed the ``result`` step from the outline, as well as the ``result`` output.
+    For this work chain, we're assuming that for now we are only interested in whether or not the result is even.
 
-    result = add_and_multiply(Int(2), Int(3), Int(5))
+We can avoid some code duplication by simply submitting the ``MultiplyAddWorkChain`` within one of the steps of our new work chain.
+In order to do this, we replace the ``multiply``, ``add`` and ``validate_result`` steps in the outline with one ``multiply_add`` step, in which we ``submit`` the ``MultiplyAddWorkChain`` and add the returned ``WorkChainNode`` to the context:
 
-Here, the ``add_and_multiply`` work function returns the output ``Int`` node and we assign it to the variable ``result``.
-Note that the input arguments of a work function must be an instance of ``Data`` node, or any of its subclasses.
-Just calling the ``add_and_multiply`` function with regular integers will result in a ``ValueError``, as these cannot be stored in the provenance graph.
+.. literalinclude:: include/snippets/extend_workflows.py
+    :language: python
+    :pyobject: BetterMultiplyAddIsEvenWorkChain
 
-.. note::
+This already simplifies the extended work chain, and avoids duplicating the steps of the ``MultiplyAddWorkChain`` in the outline.
+However, we still had copy all of the input definitions of the ``MultiplyAddWorkChain``, and manually extract them from the inputs before passing them to the ``self.submit`` method.
+Fortunately, there is a better way of *exposing* the inputs and outputs of subprocesses of the work chain.
 
-    Although the example above shows the most straightforward way to run the ``add_and_multiply`` work function, there are several other ways of running processes that can return more than just the result.
-    For example, the ``run_get_node`` function from the AiiDA engine returns both the result of the workflow and the work function node.
-    See the :ref:`corresponding topics section for more details <topics:processes:usage:launching>`.
+Exposing inputs and outputs
+---------------------------
 
-Work chains
------------
+Any work chain *needs* to *expose* the inputs of the subprocess it wraps, and perhaps *wants* to do the same for the outputs it produces, although the latter is not necessary.
```

Comment:
Fair point! After all, `is_even` is a subprocess of the work chain, and we do not expose its inputs. Let me see if I can rewrite this so it's still clear which inputs need exposing.

Reply:
Not sure how to explain this without going into an excessive tangent discussion. Will just make the sentence less _needy_.

Reply:
mmmm, maybe something like:

```
 In many cases it is convenient for workchains to expose the inputs of the subprocess it wraps so that users can cutomize these runs directly, as well as exposing some of the outputs produced as the final results of the whole procedure.
 ```